common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -227,9 +237,113 @@ public boolean onTouch(final View view, final MotionEvent event) {
         mWebView.getSettings().setBuiltInZoomControls(webViewZoomControlsEnabled);
         mWebView.getSettings().setSupportZoom(webViewZoomEnabled);
         mWebView.setVisibility(View.INVISIBLE);
+        mWebView.getSettings().setPluginState(WebSettings.PluginState.ON);
         mWebView.setWebViewClient(webViewClient);
+        mWebView.setWebChromeClient(new WebChromeClient() {
+            @Override
+            public void onPermissionRequest(final PermissionRequest request) {
+                // We can only grant or deny permissions for video capture/camera.
+                // To avoid unintentionally granting requests for not defined permissions.
+                if (iSPermissionRequestForCamera(request)) {
+                    mCameraPermissionRequest = request;
+                    if (isCameraPermissionGranted()) {
+                        acceptCameraRequest();
+                    } else if (shouldShowRequestPermissionRationale(Manifest.permission.CAMERA)) {
+                        showCameraRationale();
```

Comment:
after requestCameraPermission(), do we have to invoke acceptCameraRequest()?

Reply:
no, the user can deny or allow the permission. this is handled by the cameraRequestActivity